src/Features/CSharp/Portable/CodeFixes/AddExplicitCast/AddExplicitCastCodeFixProvider.cs

```diff
@@ -0,0 +1,380 @@
+        /// 
+        /// <param name="argumentList"> The argument list of invocation expression</param>
+        /// <param name="parameters"> The parameters of function</param>
+        /// <param name="targetArgument"> The target argument that contains target node</param>
+        /// <param name="targetArgumentConversionType"> Output the corresponding parameter type of
+        /// the target arugment if function returns true</param>
+        /// <returns>
+        /// True, if arguments and parameters match perfectly. <paramref name="targetArgumentConversionType"/> Output the corresponding parameter type
+        /// False, otherwise.
+        /// </returns>
+        private static bool CanArgumentTypesBeConvertedToParameterTypesTest(
+            SemanticModel semanticModel, SyntaxNode root, ArgumentListSyntax argumentList,
+            ImmutableArray<IParameterSymbol> parameters, ArgumentSyntax targetArgument,
+            CancellationToken cancellationToken, [NotNullWhen(true)] out ITypeSymbol? targetArgumentConversionType)
+        {
+            targetArgumentConversionType = null;
+
+            // No conversion happens under this case
+            if (parameters.Length == 0)
+                return false;
+
+            var arguments = argumentList.Arguments;
+            var newArguments = new List<ArgumentSyntax>();
+
+            for (var i = 0; i < arguments.Count; i++)
+            {
+                // Parameter index cannot out of its range, #arguments is larger than #parameter only if the last parameter with keyword params
+                var parameterIndex = Math.Min(i, parameters.Length - 1);
+
+                // If the argument has a name, get the corresponding parameter index
+                var nameSyntax = arguments[i].NameColon?.Name;
+                if (nameSyntax != null)
+                {
+                    var name = nameSyntax.Identifier.ValueText;
+                    if (!FindCorrespondingParameterByName(name, parameters, ref parameterIndex))
+                        return false;
+                }
+
+                // The argument is either in order with parameters, or have a matched name with parameters.
+                // "argumentType.Type" could be null when "augumentExpression" is lambada expression.
+                var augumentExpression = arguments[i].Expression;
+                var argumentType = semanticModel.GetTypeInfo(augumentExpression, cancellationToken);
+                var parameterType = parameters[parameterIndex].Type;
+
+                if (parameters[parameterIndex].IsParams
+                    && parameters.Last().Type is IArrayTypeSymbol paramsType
+                    && (semanticModel.ClassifyConversion(augumentExpression, paramsType.ElementType).Exists
+                    || (argumentType.Type is object
+                    && semanticModel.Compilation.ClassifyCommonConversion(argumentType.Type, paramsType.ElementType).Exists)))
```

Comment:
this expression is far too complex.  Please extract a method and document things.

Note: ti don't know why you call both `semanticModel.ClassifyConversion(augumentExpression, paramsType.ElementType)` *and* `semanticModel.Compilation.ClassifyCommonConversion(argumentType.Type, paramsType.ElementType)`.  These decisions need to be doc'ed.

Reply:
Actually `semanticModel.Compilation.ClassifyCommonConversion` is only for arguments with the keyword `out`. For example, 
```csharp
void Foo(int i, out int j) { j = 1; } 

Foo(1, out var j);
```
`semanticModel.ClassifyConversion("var j", "int")` doesn't work because the kind of "var j" is `Declaration Expression` and conversion from a declaration is unspecified.

I am not sure if there is a better way to handle this case.

Reply:
I leave it in a separated case:)